android/sdk/src/main/java/org/jitsi/meet/sdk/connection_service/ConnectionImpl.java

```diff
@@ -0,0 +1,98 @@
+package org.jitsi.meet.sdk.connection_service;
+
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+import android.telecom.Connection;
+import android.telecom.PhoneAccountHandle;
+import android.telecom.TelecomManager;
+
+import com.facebook.react.bridge.WritableNativeMap;
+
+import org.jitsi.meet.sdk.ReactContextUtils;
+
+/**
+ * Connection implementation for Jitsi Meet's {@link ConnectionService}.
+ *
+ * @author Pawel Domas
+ */
+@RequiresApi(api = Build.VERSION_CODES.O)
+public class ConnectionImpl extends Connection {
+
+    private final ConnectionService service;
+
+    ConnectionImpl(ConnectionService service) {
+        this.service = service;
+    }
+
+    /**
+     * Called when system wants to disconnect the call.
+     *
+     * {@inheritDoc}
+     */
+    @Override
+    public void onDisconnect() {
+        WritableNativeMap data = new WritableNativeMap();
+        data.putString("callUUID", getCallUUID());
+        ReactContextUtils.emitEvent(
+                    null,
+                    "org.jitsi.meet:features/connection_service#disconnect",
+                    data);
+    }
+
+    /**
+     * Called when system wants to abort the call.
+     *
+     * {@inheritDoc}
+     */
+    @Override
+    public void onAbort() {
+        WritableNativeMap data = new WritableNativeMap();
+        data.putString("callUUID", getCallUUID());
+        ReactContextUtils.emitEvent(
+                null,
+                "org.jitsi.meet:features/connection_service#abort",
+                data);
+    }
+
+    /**
+     * Unregisters the account when the call is disconnected.
+     *
+     * @param state - the new connection's state.
+     */
+    @Override
+    public void onStateChanged(int state) {
+        super.onStateChanged(state);
+
+        if (state == STATE_DISCONNECTED) {
```

Comment:
Don't we need to emit any event here?

Reply:
No, it goes the other way around. When onDisconnect is called on the ConnectionImpl it emits "org.jitsi.meet:features/connection_service#disconnect" and then goes back from JS to Java through endCall in RNConnectionService (which was missing, sorry I don't know how it happened it should be there now).

Reply:
👍